[Bug]: Can't compile rust -> error: linking with `cc` failed: exit status: 1

### Problem description

Can't compile rust -> error: linking with `cc` failed: exit status: 1

https://github.com/sviete/AIS-linux-packages/runs/7628764337?check_suite_focus=true

### What steps will reproduce the bug?

```
./build-package.sh -a arm rust
```

```
Compiling toml v0.5.7
error: linking with `cc` failed: exit status: 1
  |
  = note: "cc" "-m64" "/tmp/rustcFXTkRP/symbols.o" "/home/builder/.termux-build/rust/build/build/bootstrap/debug/deps/bootstrap-f666a6f9af3a90e9.1bhyaz7pz2v984yd.rcgu.o" "/home/builder/.termux-build/rust/build/build/bootstrap/debug/deps/bootstrap-f666a6f9af3a90e9.1srrrn682ee3ijqt.rcgu.o" "/home/builder/.termux-build/rust/build/build/bootstrap/debug/deps/bootstrap-f666a6f9af3a90e9.1w7ks4vzzr6dsjgk.rcgu.o" "/home/builder/.termux-build/rust/build/build/bootstrap/debug/deps/bootstrap-f666a6f9af3a90e9.2c8a3ds9yidsz3ri.rcgu.o" "/home/builder/.termux-build/rust/build/build/bootstrap/debug/deps/bootstrap-f666a6f9af3a90e9.2pelcrx5sjfqm7dt.rcgu.o" "/home/builder/.termux-build/rust/build/build/bootstrap/debug/deps/bootstrap-f666a6f9af3a90e9.34odv08emyabu2dh.rcgu.o" "/home/builder/.termux-build/rust/build/build/bootstrap/debug/deps/bootstrap-f666a6f9af3a90e9.3et47m9csply5emm.rcgu.o" "/home/builder/.termux-build/rust/build/build/bootstrap/debug/deps/bootstrap-f666a6f9af3a90e9.3vu5clboooahbkdc.rcgu.o" "/home/builder/.termux-build/rust/build/build/bootstrap/debug/deps/bootstrap-f666a6f9af3a90e9.3xqucwbt6kw3bc8j.rcgu.o" "/home/builder/.termux-build/rust/build/build/bootstrap/debug/deps/bootstrap-f666a6f9af3a90e9.48hyvejpcueqsn75.rcgu.o" "/home/builder/.termux-build/rust/build/build/bootstrap/debug/deps/bootstrap-f666a6f9af3a90e9.4b2cs872se0rle9k.rcgu.o" "/home/builder/.termux-build/rust/build/build/bootstrap/debug/deps/bootstrap-f666a6f9af3a90e9.4cnmwvci2wnqguhz.rcgu.o" "/home/builder/.termux-build/rust/build/build/bootstrap/debug/deps/bootstrap-f666a6f9af3a90e9.4l6mffsglu3k7p1i.rcgu.o" "/home/builder/.termux-build/rust/build/build/bootstrap/debug/deps/bootstrap-f666a6f9af3a90e9.4vxj0djuqvpkyo3r.rcgu.o" "/home/builder/.termux-build/rust/build/build/bootstrap/debug/deps/bootstrap-f666a6f9af3a90e9.50qohwpjtd154fhy.rcgu.o" "/home/builder/.termux-build/rust/build/build/bootstrap/debug/deps/bootstrap-f666a6f9af3a90e9.6hlab4apvroem44.rcgu.o" "/home/builder/.termux-build/rust/build/build/bootstrap/debug/deps/bootstrap-f666a6f9af3a90e9.76elnh89q8lz1bw.rcgu.o" "/home/builder/.termux-build/rust/build/build/bootstrap/debug/deps/bootstrap-f666a6f9af3a90e9.rnq1702lrawtbr.rcgu.o" "/home/builder/.termux-build/rust/build/build/bootstrap/debug/deps/bootstrap-f666a6f9af3a90e9.2buy268bxe4w289n.rcgu.o" "-Wl,--as-needed" "-L" "/home/builder/.termux-build/rust/build/build/bootstrap/debug/deps" "-L" "/data/data/pl.sviete.dom/files/usr/lib" "-L" "/home/builder/.rustup/toolchains/1.62.0-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib" "-Wl,-Bstatic" "/home/builder/.termux-build/rust/build/build/bootstrap/debug/deps/libbootstrap-d851a3b949f12898.rlib" "/home/builder/.termux-build/rust/build/build/bootstrap/debug/deps/libcmake-9c7ab7dae840e70b.rlib" "/home/builder/.termux-build/rust/build/build/bootstrap/debug/deps/libtar-25275a31896838c1.rlib" "/home/builder/.termux-build/rust/build/build/bootstrap/debug/deps/libxattr-0b3315df6c46339b.rlib" "/home/builder/.termux-build/rust/build/build/bootstrap/debug/deps/libopener-9318a9a28bc43163.rlib" "/home/builder/.termux-build/rust/build/build/bootstrap/debug/deps/libnum_cpus-08497b44d82f8b64.rlib" "/home/builder/.termux-build/rust/build/build/bootstrap/debug/deps/libtoml-51d5235fc05d0ad4.rlib" "/home/builder/.termux-build/rust/build/build/bootstrap/debug/deps/libserde_json-c36820aeb05ca67c.rlib" "/home/builder/.termux-build/rust/build/build/bootstrap/debug/deps/libryu-819d1ba8d2191a19.rlib" "/home/builder/.termux-build/rust/build/build/bootstrap/debug/deps/libitoa-91caa7a99f1ad3f2.rlib" "/home/builder/.termux-build/rust/build/build/bootstrap/debug/deps/libcc-b653a1d5c3eb641b.rlib" "/home/builder/.termux-build/rust/build/build/bootstrap/debug/deps/libxz2-f4a385fbca40d037.rlib" "/home/builder/.termux-build/rust/build/build/bootstrap/debug/deps/liblzma_sys-3d6f1f173e975eaa.rlib" "/home/builder/.termux-build/rust/build/build/bootstrap/debug/deps/libignore-57dc93233992ba98.rlib" "/home/builder/.termux-build/rust/build/build/bootstrap/debug/deps/libcrossbeam_utils-20739855355590e9.rlib" "/home/builder/.termux-build/rust/build/build/bootstrap/debug/deps/libwalkdir-92198e1ce72af23c.rlib" "/home/builder/.termux-build/rust/build/build/bootstrap/debug/deps/libthread_local-0b9e37bf577e6dfb.rlib" "/home/builder/.termux-build/rust/build/build/bootstrap/debug/deps/libsame_file-079605b7befa462a.rlib" "/home/builder/.termux-build/rust/build/build/bootstrap/debug/deps/libglobset-58f2401dd4df51a7.rlib" "/home/builder/.termux-build/rust/build/build/bootstrap/debug/deps/libregex-e03a29a53fa57a35.rlib" "/home/builder/.termux-build/rust/build/build/bootstrap/debug/deps/libregex_syntax-afcb317e1bd1c8d0.rlib" "/home/builder/.termux-build/rust/build/build/bootstrap/debug/deps/liblog-2abab5d92aabd17d.rlib" "/home/builder/.termux-build/rust/build/build/bootstrap/debug/deps/libfnv-f0ed8378c4a7fe08.rlib" "/home/builder/.termux-build/rust/build/build/bootstrap/debug/deps/libbstr-b8623dec084ffd38.rlib" "/home/builder/.termux-build/rust/build/build/bootstrap/debug/deps/libregex_automata-97ca80893798a9ec.rlib" "/home/builder/.termux-build/rust/build/build/bootstrap/debug/deps/liblazy_static-d2571ad8922e7fbd.rlib" "/home/builder/.termux-build/rust/build/build/bootstrap/debug/deps/libaho_corasick-5b0daee36e894279.rlib" "/home/builder/.termux-build/rust/build/build/bootstrap/debug/deps/libmemchr-0c71a92db0f46a8d.rlib" "/home/builder/.termux-build/rust/build/build/bootstrap/debug/deps/libgetopts-0b04c15b9de51aa0.rlib" "/home/builder/.termux-build/rust/build/build/bootstrap/debug/deps/libunicode_width-35c59af2f7282ef0.rlib" "/home/builder/.termux-build/rust/build/build/bootstrap/debug/deps/libserde-63d5953176c59d55.rlib" "/home/builder/.termux-build/rust/build/build/bootstrap/debug/deps/libonce_cell-dc392680a5f69919.rlib" "/home/builder/.termux-build/rust/build/build/bootstrap/debug/deps/libfiletime-c6b80b5a6540d3c3.rlib" "/home/builder/.termux-build/rust/build/build/bootstrap/debug/deps/liblibc-0e0772a3a7def9e7.rlib" "/home/builder/.termux-build/rust/build/build/bootstrap/debug/deps/libcfg_if-3cca6bc90a87dbab.rlib" "-Wl,--start-group" "/home/builder/.rustup/toolchains/1.62.0-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libstd-7ca39ac42651c3df.rlib" "/home/builder/.rustup/toolchains/1.62.0-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libpanic_unwind-62c6d032818141a1.rlib" "/home/builder/.rustup/toolchains/1.62.0-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libobject-50484fc03eb1eb5b.rlib" "/home/builder/.rustup/toolchains/1.62.0-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libmemchr-758be083b246d9c6.rlib" "/home/builder/.rustup/toolchains/1.62.0-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libaddr2line-3cdf9a3c68f76e2d.rlib" "/home/builder/.rustup/toolchains/1.62.0-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libgimli-3a1b74821c25a0e1.rlib" "/home/builder/.rustup/toolchains/1.62.0-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/librustc_demangle-e046d82ebd84bb7f.rlib" "/home/builder/.rustup/toolchains/1.62.0-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libstd_detect-a61cdd33cfa8394f.rlib" "/home/builder/.rustup/toolchains/1.62.0-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libhashbrown-410c38f8df854235.rlib" "/home/builder/.rustup/toolchains/1.62.0-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libminiz_oxide-f79d7458e122215f.rlib" "/home/builder/.rustup/toolchains/1.62.0-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libadler-7d24b750ce5b22e8.rlib" "/home/builder/.rustup/toolchains/1.62.0-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/librustc_std_workspace_alloc-4c2aa1ea3133ab73.rlib" "/home/builder/.rustup/toolchains/1.62.0-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libunwind-41220dc85a7f114f.rlib" "/home/builder/.rustup/toolchains/1.62.0-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcfg_if-230d004276c898f9.rlib" "/home/builder/.rustup/toolchains/1.62.0-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/liblibc-03ae30169a5438be.rlib" "/home/builder/.rustup/toolchains/1.62.0-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/liballoc-6f82c44b7818af35.rlib" "/home/builder/.rustup/toolchains/1.62.0-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/librustc_std_workspace_core-1a0b7681f7efa789.rlib" "/home/builder/.rustup/toolchains/1.62.0-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcore-83735dd4dae9b02c.rlib" "-Wl,--end-group" "/home/builder/.rustup/toolchains/1.62.0-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcompiler_builtins-385029872275478f.rlib" "-Wl,-Bdynamic" "-llzma" "-lgcc_s" "-lutil" "-lrt" "-lpthread" "-lm" "-ldl" "-lc" "-Wl,--eh-frame-hdr" "-Wl,-znoexecstack" "-L" "/home/builder/.rustup/toolchains/1.62.0-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib" "-o" "/home/builder/.termux-build/rust/build/build/bootstrap/debug/deps/bootstrap-f666a6f9af3a90e9" "-Wl,--gc-sections" "-pie" "-Wl,-zrelro,-znow" "-nodefaultlibs" "-Wl,-rpath=/data/data/pl.sviete.dom/files/usr/lib" "-Wl,--enable-new-dtags"
  = note: ld: error: /data/data/pl.sviete.dom/files/usr/lib/liblzma.a(liblzma_la-common.o) is incompatible with elf64-x86-64
          ld: error: /data/data/pl.sviete.dom/files/usr/lib/liblzma.a(liblzma_la-alone_encoder.o) is incompatible with elf64-x86-64
          ld: error: /data/data/pl.sviete.dom/files/usr/lib/liblzma.a(liblzma_la-easy_encoder.o) is incompatible with elf64-x86-64
          ld: error: /data/data/pl.sviete.dom/files/usr/lib/liblzma.a(liblzma_la-stream_encoder.o) is incompatible with elf64-x86-64
          ld: error: /data/data/pl.sviete.dom/files/usr/lib/liblzma.a(liblzma_la-stream_encoder_mt.o) is incompatible with elf64-x86-64
          ld: error: /data/data/pl.sviete.dom/files/usr/lib/liblzma.a(liblzma_la-alone_decoder.o) is incompatible with elf64-x86-64
          ld: error: /data/data/pl.sviete.dom/files/usr/lib/liblzma.a(liblzma_la-auto_decoder.o) is incompatible with elf64-x86-64
          ld: error: /data/data/pl.sviete.dom/files/usr/lib/liblzma.a(liblzma_la-stream_decoder.o) is incompatible with elf64-x86-64
          ld: error: /data/data/pl.sviete.dom/files/usr/lib/liblzma.a(liblzma_la-check.o) is incompatible with elf64-x86-64
          ld: error: /data/data/pl.sviete.dom/files/usr/lib/liblzma.a(liblzma_la-lz_encoder.o) is incompatible with elf64-x86-64
          ld: error: /data/data/pl.sviete.dom/files/usr/lib/liblzma.a(liblzma_la-lzma_encoder_presets.o) is incompatible with elf64-x86-64
          collect2: error: ld returned 1 exit status
          

error: could not compile `bootstrap` due to previous error
warning: build failed, waiting for other jobs to finish...
failed to run: /home/builder/.rustup/toolchains/1.62.0-x86_64-unknown-linux-gnu/bin/cargo build --manifest-path /home/builder/.termux-build/rust/src/src/bootstrap/Cargo.toml
Build completed unsuccessfully in 0:00:18


```

### What is the expected behavior?

rust compiled successfully 

### System information

termux-info:

```

```

--- a/packages/rust/build.sh
+++ b/packages/rust/build.sh
@@ -69,6 +69,11 @@ termux_step_configure() {
 	mv $TERMUX_PREFIX/lib/libz.so $TERMUX_PREFIX/lib/libz.so.tmp
 	mv $TERMUX_PREFIX/lib/liblzma.so.$LZMA_VERSION $TERMUX_PREFIX/lib/liblzma.so.tmp
 
+	# https://github.com/termux/termux-packages/issues/11427
+	# Fresh build conflict: liblzma -> rust
+	# ld: error: /data/data/com.termux/files/usr/lib/liblzma.a(liblzma_la-common.o) is incompatible with elf64-x86-64
+	mv $TERMUX_PREFIX/lib/liblzma.a $TERMUX_PREFIX/lib/liblzma.a.tmp || true
+
 	# ld: error: undefined symbol: getloadavg
 	# >>> referenced by rand.c
 	$CC $CPPFLAGS -c $TERMUX_PKG_BUILDER_DIR/getloadavg.c
@@ -101,6 +106,7 @@ termux_step_make_install() {
 	mv $TERMUX_PREFIX/lib/libz.so.1.tmp $TERMUX_PREFIX/lib/libz.so.1
 	mv $TERMUX_PREFIX/lib/libz.so.tmp $TERMUX_PREFIX/lib/libz.so
 	mv $TERMUX_PREFIX/lib/liblzma.so.tmp $TERMUX_PREFIX/lib/liblzma.so.$LZMA_VERSION
+	mv $TERMUX_PREFIX/lib/liblzma.a.tmp $TERMUX_PREFIX/lib/liblzma.a || true
 
 	ln -sf rustlib/$CARGO_TARGET_NAME/lib/*.so .
 	ln -sf $TERMUX_PREFIX/bin/lld $TERMUX_PREFIX/bin/rust-lld
@@ -119,4 +125,5 @@ termux_step_post_massage() {
 	rm -f lib/libz.so
 	rm -f lib/libz.so.1
 	rm -f lib/liblzma.so.$LZMA_VERSION
+	rm -f lib/liblzma.a
 }

